src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -16565,94 +16582,115 @@ def dt_round(
         if nonexistent != "raise":
             ErrorMessage.parameter_not_implemented_error("nonexistent", method_name)
 
-        slice_length, slice_unit = rule_to_snowflake_width_and_slice_unit(
-            rule=freq  # type:  ignore[arg-type]
-        )
-
-        if slice_unit not in SUPPORTED_DT_FLOOR_CEIL_FREQS or slice_unit == "second":
-            ErrorMessage.parameter_not_implemented_error(f"freq={freq}", method_name)
-
-        # We need to implement the algorithm for rounding half to even whenever
-        # the date value is at half point of the slice:
-        # https://en.wikipedia.org/wiki/Rounding#Rounding_half_to_even
-
-        # First, we need to calculate the length of half a slice.
-        # This is straightforward if the length is already even.
-        # If not, we then need to first downlevel the freq to a
-        # lower granularity to ensure that it is even.
+        dtype = self.index_dtypes[0] if include_index else self.dtypes[0]
+        if is_datetime64_any_dtype(dtype):
+            slice_length, slice_unit = rule_to_snowflake_width_and_slice_unit(
+                rule=freq  # type:  ignore[arg-type]
```

Comment:
let's update rule_to_snowflake_width_and_slice_unit's type hint to allow Frequency as input.

Reply:
done